src/Core/Components/Switch/FluentSwitch.razor.css

```diff
@@ -0,0 +1,3 @@
+fluent-switch[readonly] {
```

Comment:
Have you checked that the ReadOnly parameter blocks the use of the mouse and keyboard?

I think you need to apply this CSS style (on the “field” element and not on the “switch” element):
```css
fluent-field:has(fluent-switch[readonly]) {
  pointer-events: none;
}
```

And to block the keyboard, you need to set the `tabindex` on the `fluent-switch` (like done) but also on the `FluentField` component. If not, using `Tab` the user can navigate to the component and press `Space`

Reply:
Ok, css added for readonly and tabIndex on FluentField.